### ISSUE TITLE
Support Govee H5179 (Hygrometer)

### ISSUE THREAD

**Issue by pfleenor** [theengs/decoder#265]
Would love to see Govee H5179 & H5054 support added.  
I may take a stab at it when time permits if no one else is able.

Thanks

**Comment by DigiH** [comment]
Hi @pfleenor,

we'd like to include these two Govee devices, but would need the help of someone who actually has them, to get their raw data and comparison states, temperature and humidity for the H5179, and no-leak vs. leak state etc. for the H5054.

If you are willing to participate with such detail contributions we can take a stab at them together.

**P.S.:** From doing a quick search for these two devices it seems that the H5054 Water Leak Detector might not have bluetooth, and would therefore be out of scope of Decoder. Please let us know if otherwise and you can detect it with any bluetooth scanning app.

**Comment by pfleenor** [comment]
Yes, you're right, 5054 is wifi only :(

For 5179, I'd be happy to provide whatever is needed.

**Comment by DigiH** [comment]
> Yes, you're right, 5054 is wifi only :(

Actually the gateway for the leak detector is WiFi, whereas the leak detector itself is broadcasting RF 433 Mhz, and it is actually compatible with [OpenMQTTGateway](https://github.com/1technophile/OpenMQTTGateway) with the rtl_433_esp module already, being detected and reported as

`{"model":"Govee-Water","id":16538,"battery_ok":0.31,"battery_mV":2172,"event":"Battery Report","code":"b5f403e00218","mic":"PARITY","protocol":"Govee Water Leak Dectector H5054, Door Contact Sensor B5023","rssi":-86,"duration":5222253}`

`{"model":"Govee-Water","id":16538,"event":"Water Leak","code":"b5f40406575a","mic":"PARITY","protocol":"Govee Water Leak Dectector H5054, Door Contact Sensor B5023","rssi":-80,"duration":1967501}`

Thanks to @1technophile for the above confirmation.

Are you using OpenMQTTGatway or Theengs Gateway? If you could provide the received raw manufacturerdata of your H5179, along with the actual values it shows in the Govee app at the same time, then we can see how they are encoded in the manufacturerdata to include a decoder for it.

**Comment by pfleenor** [comment]
I'm not "using" either yet.  I was attempting to get Theengs Gateway going but I'll need to look at openMQTTGateway too.  I'll get this data as soon as I get it going.. may be a few days.

Thanks for the info on the leak detector.

**Comment by Milkysunshine** [comment]
> > Yes, you're right, 5054 is wifi only :(
> 
> Actually the gateway for the leak detector is WiFi, whereas the leak detector itself is broadcasting RF 433 Mhz, and it is actually compatible with [OpenMQTTGateway](https://github.com/1technophile/OpenMQTTGateway) with the rtl_433_esp module already, being detected and reported as
> 
> `{"model":"Govee-Water","id":16538,"battery_ok":0.31,"battery_mV":2172,"event":"Battery Report","code":"b5f403e00218","mic":"PARITY","protocol":"Govee Water Leak Dectector H5054, Door Contact Sensor B5023","rssi":-86,"duration":5222253}`
> 
> `{"model":"Govee-Water","id":16538,"event":"Water Leak","code":"b5f40406575a","mic":"PARITY","protocol":"Govee Water Leak Dectector H5054, Door Contact Sensor B5023","rssi":-80,"duration":1967501}`
> 
> Thanks to @1technophile for the above confirmation.
> 
> Are you using OpenMQTTGatway or Theengs Gateway? If you could provide the received raw manufacturerdata of your H5179, along with the actual values it shows in the Govee app at the same time, then we can see how they are encoded in the manufacturerdata to include a decoder for it.

I have 3 of the H5179 as well as a H5174. The H5174 works flawlessly with Theengs Gateway running on a pi zero w, but the 3 H5179 never show up as found.

I'll gladly provide anything needed if you give me a little direction of what is needed?

**Comment by Milkysunshine** [comment]
Looking at the debug logs, I see what I need to get. I'll have to pull the batteries out of 2 of my units to isolate the mac address of the unit I'm actually getting the info from. Govee isn't nice about revealing the mac information. 

I had to create an AP on my phone, and temporarily connect the 3 units to it in order to get their wifi mac address, which doesn't help at all with bluetooth.

**Comment by DigiH** [comment]
Hi @Milkysunshine 

Could you start Theengs Gateway with [Publish advertising and advanced data](https://gateway.theengs.io/use/use.html#details-options) set to 1 and then provide the undecoded raw data along with the readings of the H5179?

Maybe just have one active for this testing as to know which one will be showing up.

You could also try testing the initial H5179 test branch at

https://github.com/DigiH/decoder/tree/H5179

following the Theengs Gateway test decoder instructions along the lines of the Oral-B test decoder at

https://github.com/theengs/decoder/discussions/432#discussioncomment-7381669

**Comment by Milkysunshine** [comment]
I tried to let the unit stabilize in my office and get readings from around the same time. 

Temp 68.5F ( 20.3C )
Hum 57.3%
Dew Point 52.9.1F ( 11.6C )
VPD 1.02kPa

```
DEBUG:BLEGateway:E0:13:D4:11:48:31:AdvertisementData(local_name='Govee_H5179_4831', manufacturer_data={34817: b'\xec\x00\x01\x01\xee\x07X\x16A'}, service_uuids=['0000180a-0000-1000-8000-00805f9b34fb', '0000fef5-0000-1000-8000-00805f9b34fb', '0000ec88-0000-1000-8000-00805f9b34fb'], rssi=-34)
DEBUG:BLEGateway:Sent `{"manufacturerdata": "0188ec000101ee07581641", "name": "Govee_H5179_4831", "id": "E0:13:D4:11:48:31", "rssi": -34}` to topic `home/TheengsGateway/BTtoMQTT/E013D4114831`
```

**Comment by DigiH** [comment]
Thanks a lot for this @Milkysunshine, I had a small error with the actual manufacturerdata data length in my assumed test decoder, but with your data this has now been fixed, and the results for your above raw data is

`{"brand":"Govee","model":"Thermo-Hygrometer","model_id":"H5179","type":"THB","cidc":false,"acts":true,"tempc":20.3,"tempf":68.54,"hum":57.2,"batt":65}`

Please note that the Dew Point and VPD are not being transmitted by the actual device, but they are being calculated within the app, same as can be done in any controller with the existing temperature and humidity data received.

**Comment by Milkysunshine** [comment]
> Please note that the Dew Point and VPD are not being transmitted by the actual device, but they are being calculated within the app, same as can be done in any controller with the existing temperature and humidity data received.

I assumed that was the case as it is easily calculated.

No worries on the info.

How do I go about implementing this?

**Comment by DigiH** [comment]
I'll merge this into the development branch of Decoder now so it will be included in the next Decoder release, which will then be in the next Gateway release.

If you would want this immediately you'd have to make your own build of Decoder from the development branch, as described at

https://decoder.theengs.io/use/python.html#installing-a-development-version

**Comment by Milkysunshine** [comment]
I guess I'll have to wait, as trying to build it gave an error, which I don't have the time or desire to diagnose right now... especially on a pi zero w. Nothing happens fast on that thing!

**Comment by DigiH** [comment]
Closing as the H5179 decoder now being included in Theengs Decoder release 1.6.0.

@Milkysunshine There should be a Theengs Gateway update shortly, but if you uninstall and reinstall your current release of Theengs Gateway it should also pull in the newer Decoder.
